routes/login.js

```diff
@@ -0,0 +1,142 @@
+                        res.status(500).send(
+                            new Error("INTERNAL_SERVER_ERROR", error)
+                        );
+                    } else {
+                        if (match) {
+                            const token = jwtService.sign(user.department);
+                            token != null
+                                ? res.status(200).json({ token: token })
+                                : res
+                                      .status(500)
+                                      .send(
+                                          new Error(
+                                              "INTERNAL_SERVER_ERROR",
+                                              "Internal error occurred!"
+                                          )
+                                      );
+                        } else {
+                            res.status(401).send(
+                                new Error("UNAUTHORIZED", "Password incorrect!")
+                            );
+                        }
+                    }
+                });
+            })
+            .catch(() => {
+                res.status(400).send(
+                    new Error("BAD_REQUEST", "User not found!")
+                );
+            });
+    }
+});
+
+router.post("/add", (req, res) => {
+    const { zprn, firstName, lastName, password, department, role } = req.body;
+    const { er, value } = validateData(
+        zprn,
+        firstName,
+        lastName,
+        password,
+        department,
+        role
+    );
+    if (er != undefined) {
+        res.status(400).send(
+            new Error("BAD_REQUEST", error.details[0].message)
+        );
+    } else {
+        bcrypt.genSalt(10, (e, salt) => {
+            if (!e) {
+                bcrypt.hash(password, salt, (error, hash) => {
+                    if (error) {
+                        res.status(500).send(
+                            new Error("INTERNAL_SERVER_ERROR", error)
+                        );
+                    } else {
+                        if (!hash) {
+                            res.status(500).send(
+                                new Error(
+                                    "INTERNAL_SERVER_ERROR",
+                                    "An internal error occured!"
+                                )
+                            );
+                        } else {
+                            userService
+                                .addUser(
+                                    zprn,
+                                    firstName,
+                                    lastName,
+                                    hash,
+                                    department,
+                                    role
+                                )
+                                .then(() => {
+                                    res.status(20).json({
+                                        code: "CREATED",
```

Comment:
Shouldn't `code:` be a number ?  A status code as you have defined in the `class Error`. Rather I think we should add a `result` to this to save the `USER_CREATED`  instead of `CREATED` ?

Reply:
The link you shared stated code as String, because the response status is explicitly set. I think we should go with this.

Reply:
The link you shared stated code as String, as the status code is explicitly set in the response. Hence, I think we should go with this.

Reply:
Okay makes sense, as we already have a status in the HTTP header.